src/components/exercise/index.cjsx

```diff
@@ -122,16 +120,14 @@ ExerciseWithScroll = React.createClass
 
   componentDidMount: ->
     {currentStep} = @props
-    if currentStep?
-      @scrollToSelector("[data-step='#{currentStep}']")
+    @scrollToStep(currentStep) if currentStep?
 
   componentWillReceiveProps: (nextProps) ->
-    if nextProps.currentStep isnt @props.currentStep
-      @scrollToSelector("[data-step='#{nextProps.currentStep}']")
+    @scrollToStep(nextProps.currentStep) if nextProps.currentStep isnt @props.currentStep
 
-  onAfterScroll: ->
-    textArea = @_scrollingTargetDOM().querySelector("[data-step='#{@props.currentStep}'] textarea")
-    textArea?.focus()
+  scrollToStep: (currentStep) ->
+    stepSelector = "[data-step='#{currentStep}']"
+    @scrollToSelector(stepSelector, updateHistory: false) unless @isSelectorInView(stepSelector)
 
```

Comment:
I like this pattern - since `scrollToSelector` already accepts an options object, should we add a `onlyIfHidden` option so we could call this as: @scrollToSelector(stepSelector, updateHistory: false, onlyIfHidden: true)` ?

Of course, you can probably think up a bitter option name :)